src/expressions.md

```diff
@@ -8,52 +8,92 @@ several things:
+  expression
 
 In this way, the structure of expressions dictates the structure of execution.
 Blocks are just another kind of expression, so blocks, statements, expressions,
 and blocks again can recursively nest inside each other to an arbitrary depth.
 
-### Lvalues, rvalues and temporaries
+## Lvalues and rvalues
 
 Expressions are divided into two main categories: _lvalues_ and _rvalues_.
 Likewise within each expression, sub-expressions may occur in _lvalue context_
 or _rvalue context_. The evaluation of an expression depends both on its own
 category and the context it occurs within.
 
 An lvalue is an expression that represents a memory location. These expressions
-are [paths](#path-expressions) (which refer to local variables, function and
-method arguments, or static variables), dereferences (`*expr`), [indexing
-expressions](#index-expressions) (`expr[expr]`), and [field
-references](#field-expressions) (`expr.f`). All other expressions are rvalues.
+are [paths](#path-expressions) which refer to local variables, function and
+method arguments, or static variables,
+[dereferences](#the-dereference-operator) (`*expr`), [indexing
+expressions](#index-expressions) (`expr[expr]`), [field
+references](#field-expressions) (`expr.f`) and parenthesized lvalue
+expressions. All other expressions are rvalues.
 
 The left operand of an [assignment](#assignment-expressions) or
-[compound-assignment](#compound-assignment-expressions) expression is
-an lvalue context, as is the single operand of a unary
-[borrow](#unary-operator-expressions). The discriminant or subject of
-a [match expression](#match-expressions) may be an lvalue context, if
-ref bindings are made, but is otherwise an rvalue context. All other
-expression contexts are rvalue contexts.
+[compound-assignment](#compound-assignment-expressions) expression is an lvalue
+context, as is the single operand of a unary [borrow](#borrow-operators), and
+the operand of any [implicit borrow](#implicit-borrows). The discriminant or
+subject of a [match expression](#match-expressions) and right side of a `let`
+binding may be an lvalue context, if ref bindings are made, but is otherwise an
+rvalue context. All other expression contexts are rvalue contexts.
 
-When an lvalue is evaluated in an _lvalue context_, it denotes a memory
-location; when evaluated in an _rvalue context_, it denotes the value held _in_
-that memory location.
-
-#### Temporary lifetimes
+### Moved and copied types
 
-When an rvalue is used in an lvalue context, a temporary un-named
-lvalue is created and used instead. The lifetime of temporary values
-is typically the innermost enclosing statement; the tail expression of
-a block is considered part of the statement that encloses the block.
+When an lvalue is evaluated in an _rvalue context_, it denotes the value held
+_in_ that memory location. If value is of a type that implements `Copy`, then
+the value will be copied. In other situations if the type of the value is
```

Comment:
There are a few places I'm using 'other' to mean 'all other' or 'otherwise'. Any suggestions for an alternative wording would also be appreciated.